test/cypress/integration/pages/consumer-complaints.js

```diff
@@ -0,0 +1,75 @@
+import { ConsumerComplaints } from '../../pages/data-research/consumer-complaints';
+
+const page = new ConsumerComplaints();
+const states = ['AK', 'AL', 'AZ', 'AR', 'CA', 'CO', 'CT', 'DE', 'FL', 'GA',
+  'HI', 'ID', 'IL', 'IN', 'IA', 'KS', 'KY', 'LA', 'ME', 'MD', 'MA', 'MI', 'MN',
+  'MS', 'MO', 'MT', 'NE', 'NV', 'NH', 'NJ', 'NM', 'NY', 'NC', 'ND', 'OH', 'OK',
+  'OR', 'PA', 'RI', 'SC', 'SD', 'TN', 'TX', 'UT', 'VT', 'VA', 'WA', 'WV', 'WI', 'WY'];
+
+describe( 'Consumer Complaint Database', () => {
+  beforeEach( () => {
+    cy.visit( '/data-research/consumer-complaints/' );
+  } );
+
+  it( 'should display chart with complaints for all 50 states', () => {
+    page.clickButton( 'Complaints' );
+    page.checkLegend( 'description' ).should( 'contain', 'Complaints' );
+    page.chartSize( 580, 940 );
```

Comment:
Because the chart will update at different viewport sizes it might be sufficient to test that the element exists and contains tiles, rather than looking for specific pixles (this fails locally with a height of 925, for example) but otherwise these look great!

Reply:
Thanks for your review comments. I will remove the test for the map size and just verify that the map is visible. Currently only the static chart is being tested. More effort would be required to get the elasticsearch for CCDB functional in Docker where the functional tests are being run.